demo/admin/src/products/categories/ProductCategoriesPage.tsx

```diff
@@ -45,7 +56,69 @@ const ProductCategoriesPage: React.FC = () => {
                     {(selectedId) => (
                         <SaveBoundary>
                             <FormToolbar />
-                            <ProductCategoryForm id={selectedId} />
+                            <MainContent fullHeight>
+                                <RouterTabs>
+                                    <RouterTab
+                                        forceRender={true}
+                                        path=""
+                                        label={intl.formatMessage({ id: "products.editProductCategory.formTab", defaultMessage: "Product category" })}
+                                    >
+                                        <ProductCategoryForm id={selectedId} />
+                                    </RouterTab>
+                                    <RouterTab
+                                        forceRender={true}
+                                        path="/assigned-products"
+                                        label={intl.formatMessage({
+                                            id: "products.editProductCategory.assignedProducts",
+                                            defaultMessage: "Assigned Products",
+                                        })}
+                                    >
+                                        <Box sx={{ height: "100vh" }}>
+                                            <ProductsGrid
+                                                // TODO refresh ProductsGrid after saving assign-products
+                                                toolbarAction={
+                                                    <Button
+                                                        startIcon={<AddIcon />}
+                                                        onClick={() => setIsOpen(true)}
+                                                        variant="contained"
+                                                        color="primary"
+                                                    >
+                                                        <FormattedMessage
+                                                            id="products.editProductCategory.assignProducts"
+                                                            defaultMessage="Assign Products"
+                                                        />
+                                                    </Button>
+                                                }
+                                                filter={{ category: { equal: selectedId } }}
+                                            />
+                                            <SaveBoundary onAfterSave={() => setIsOpen(false)}>
```

Comment:
Please move this (ProductsGrid+SaveBoundary+Dialog etc) into it's own Component (and file)

ProductCateogriesPage should only combine together components and not have any logic on it's own.

Reply:
I'd rather only move SaveBoundary+Dialog into a SaveDialog-component (EditDialog does not work in this case). Because I would expect ProductCateogriesPage to decide how the structure+navigation is by combining ProductsGrid + AssignProductsGrid (by wrapping in SaveDialog) in ProductCateogriesPage. (maybe also wrap ProductsGrid in AssignedProductsGrid-component, to make it more obvious and hide the filtering away. If it wasn't this weird ManyToOne-situation there would probably be already a generated AssignedProductsGrid component)

additionally the page-component could be further split if it becomes to nested... (e.g. ProductCategoryDetailPage, ProductCategoryAssignedProductsRouterTab)